settings/sample-secrets.settings.php

```diff
@@ -7,34 +7,25 @@
  * NOTE: Do not edit this file directly, the contents are not intended to be
  * stored in version control. Copy file to the hosted environment, e.g.:
  *
- *   /mnt/files/{$sitegroup}.{$env}/secrets.settings.php
+ *   /mnt/gfs/home/{$sitegroup}/secrets.settings.php
  *
- * Make changes to the file as needed there.
+ * Make changes to the file as needed there. Note that this file will need to be
+ * created once for each file server in your application (typically one for
+ * dev/stage and one for prod).
  */
 
-// Ensure that Acquia Hosting database settings exist.
-if (isset($conf['acquia_hosting_site_info']['db'])) {
-
-  /*
-   * Some variables we can switch on:
-   * - ['db']['role'] is specific to a Drupal site installation.
-   * - ['db']['name'] is specific to a Drupal site in a specific environment.
-   */
-  switch ($conf['acquia_hosting_site_info']['db']['name']) {
-    case 'blted':
-      /*
-       * Acquia Network Connector
-       *
-       * @see https://docs.acquia.com/network/install
-       * @see https://insight.acquia.com/subscriptions/keys
-       *
-       * @codingStandardsIgnoreStart
-       *
-       * $config['acquia_connector.settings']['identifier'] = 'ABCD-12345';
-       * $config['acquia_connector.settings']['key'] = '00000000000000000000000000000000';
-       *
-       * @codingStandardsIgnoreEnd
-       */
-      break;
-  }
+/**
+ * The below example would set a sensitive API key for the "foo" module.
+ *
+ * In addition to $ah_env, you can use other variables defined in blt.settings.php
+ * such as $is_dev_env, $is_stage_env, $is_prod_env, and $ah_site.
+ */
+switch ($ah_env) {
+  case 'dev':
+  case 'test':
+    $settings['foo']['super_secret_key'] = 'DEV-1234';
```

Comment:
Is this any different than?
```
$config['foo.settings']['super_secret_key']
```

Reply:
Settings and config are different subsystems. Whether you should use `$settings` or `$config` depends on how exactly the module you are using stores this data.

Well-designed modules will use settings to store sensitive data like API keys, rather than config. Config is designed to be portable, exportable, stored in VCS, etc... and API keys should be none of those things.

ref: https://anavarre.net/from-conf-to-config-and-settings-in-drupal-8/#whentousesettingsindrupal8